src/util/RandHasher.h

```diff
@@ -0,0 +1,24 @@
+// Copyright 2020 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#pragma once
+#include "util/Math.h"
+#include <functional>
+#include <limits>
+
+namespace std
+{
+template <class T, class Hasher = std::hash<T>> class RandHasher
+{
+  public:
+    size_t
+    operator()(T const& t) const
+    {
+        static size_t mixer =
```

Comment:
I'm not sure it's ok to be using a static local for this -- you could race with another thread on initialization and get a hashtable that uses a half-written `mixer` value or a `mixer` that's initialized twice.